index.html

```diff
@@ -2209,218 +2270,232 @@ <h3><dfn>Writing or pushing content</dfn></h3>
+                  Set <a>pending push tuple</a> to (`this`, |p|).
+                </li>
+                <li>
+                  Run the <a>start the push timer</a> steps.
+                </li>
+                <li>
+                  Run the <a>add write handler</a> steps.
+                </li>
+                <p class="note">
+                  If <a>NFC is suspended</a>,
+                  continue waiting until |timer| expires (if set), or
+                  promise is aborted by the user, or until an <a>NFC device</a>
+                  comes within communication range.
+                </p>
+              </ol>
+            </li>
+          </li>
+        </ol>
+      </p>
+
+      <p id="steps-push-timer">
+        To <dfn>start the push timer</dfn>, run these steps:
+        <ol class=algorithm>
+          <p>
+            Let |p:Promise| be the <a>pending push tuple</a>'s promise.
+          </p>
+          <li>
+            Let |writer| be the <a>pending push tuple</a>'s writer.
+          </li>
+          <li>
+            Let |timeout:number| be |writer|.[[\PushOptions]]'s timeout.
+            <ul>
               <li>
                 If |timeout| value is `NaN` or negative, reject
-                |p| with a {{TypeError}} and abort
-                these steps.
+                |p| with a {{TypeError}} and abort these steps.
               </li>
               <li>
                 If |timeout| value is not supported by the UA, reject
                 |p| with a {{"NotSupportedError"}}
                 {{DOMException}} and abort these steps.
               </li>
               <li>
-                Let |output| be the notation for the <a>NDEF message</a>
-                to be created by UA, as the result of passing
-                |message| to <a>create Web NFC message</a>.
-                If this throws an exception, reject |p| with that
-                exception and abort these steps.
-              </li>
-              <li>
-                If |target| is "`any`", run the following
-                steps twice, once with |slot| set to the value
-                "`tag`", and once set to the value "`peer`";
-                otherwise run the following step once, with
-                |slot| set to the value of |target|.
-              </li>
-              <li>
-                Attempt to <a>abort a pending push operation</a>.
+                If |timeout| value is not equal to `Infinity`, abort these
+                steps.
               </li>
+            </ul>
+          </li>
+          <li>
+            Start a timer |timer| with the timeout value set to
+            |timeout| and set |writer|.[[\PushTimer]] to |timer|.
+          </li>
+          <li>
+            If |timer| expires, reject |p| with {{"TimeoutError"}}
+            {{DOMException}}.
+          </li>
+        </ol>
+      </p>
+
+      <p id="steps-write-handler">
+        To <dfn>add write handler</dfn>, run these steps <a>in parallel</a>:
```

Comment:
add A writer handler

Reply:
I think we need to write this a bit like the promise/React and abort steps.

![image](https://user-images.githubusercontent.com/1101339/64478201-2b6ec700-d1a5-11e9-819c-00c3d1cbaf6b.png)

Reply:
We are not adding an object or value, but a set of steps, so the internal slot probably doesn't make so much sense for this.

"Whenever an NFC device comes within communication range, execute 'write algorithms'"

Reply:
Add the following 'write algorithms' given promise p and NFCWriter writer

Reply:
OK, makes sense.